app/abtests/Test.scala

```diff
@@ -0,0 +1,54 @@
+package abtests
+
+import com.github.slugify._
+import play.api.libs.json.{JsValue, Json, Writes}
+import play.api.mvc.Cookie
+
+case class Test(name: String, audienceSize: Double, audienceOffset: Double, variants: Variant*) {
```

Comment:
There's a few cases which the `Test` logic doesn't cover e.g. 
```
val idRange = 0 to 5
val variants = ??? // List with 4 variants 
// omits two values in the id range
idRange.grouped(idRange.length / variants.size)
  .zip(variants.iterator)
```
(I'm guessing this isn't the desired behaviour?)